keyvault/client/src/main/java/com/azure/keyvault/RequestIdPolicy.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.keyvault;
+
+import com.azure.common.http.HttpPipelineCallContext;
+import com.azure.common.http.HttpPipelineNextPolicy;
+import com.azure.common.http.HttpResponse;
+import com.azure.common.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+/**
+ * Creates a policy which puts a UUID in the request header. Azure uses
+ * the request id as the unique identifier for the request.
+ * Also sets 'x-ms-return-client-request-id' header to tell server to return request id in the response.
+ */
+public class RequestIdPolicy implements HttpPipelinePolicy {
```

Comment:
As I noted with the last HttpPipelinePolicy, I need to improve my specs here, but I note that this class is public whereas the previous one was package-private. Was that intentional (it's package-private in @conniey's codebase)?

Reply:
Additionally, because this is exactly the same code as in configuration service, I suggest that you and @conniey sync up and consider elevating this to azure-common implementation API.

Reply:
Is this part of the KeyVault spec? This policy was added because users may want to keep track of their requests in Application Configuration.

Reply:
made it package-private.
Will need to move it to azure-common.
Opening a separate thread discussion (connie, you and me) about things to be moved.

Reply:
This one might not need to be moved.
Double checking if KV needs this policy.